doc/cookbook/time_windows.md

```diff
@@ -0,0 +1,189 @@
+# Processing Time-Windowed Data
+
+If you are analyzing data that is changing over time, chances are that you will want to perform some sort of analysis on "the last two weeks of data," "January's data," or some other moving or static time window of data.  There are a few different ways of doing these types of analyses in Pachyderm, depending on your use case.  We recommend one of the following patterns:
+
+1. [Fixed time window directory structures](#fixed-time-window-directory-structures) - for rigid, fixed time windows, such as months (Jan, Feb, etc.) or days (01-01-17, 01-02-17, etc.).
+
+2. [Rolling time window directory structures](#rolling-time-window-directory-structures) - for rolling time windows of data, such as three day windows or two week windows. 
+
+3. [Shuffled moving time windows](#shuffled-moving-time-windows) - for updating a constanly moving time-windowed data set.
```

Comment:
Regarding including or not including JD's method (which in my understanding is (2)), wouldn't client side hashing be relevant to either (2) or (3).  That is, it seems like the client side hashing would be relevant to either of the rolling/moving time window examples, so I think including them both with the note would be fine.  Unless there is a super clear advantage of one over the other right now.